### PR TITLE
[20331] Support Fast CDR v2

### PR DESCRIPTION
**Pull request by MiguelCompany**

### CODE REVIEW

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp [hunk 1]
```diff
@@ -301,12 +302,12 @@ const TypeObject * GetCompleteObject(
   // DDS document)
   eprosima::fastcdr::Cdr ser(
     fastbuffer, eprosima::fastcdr::Cdr::LITTLE_ENDIANNESS,
-    eprosima::fastcdr::Cdr::DDS_CDR);  // Object that serializes the data.
+    eprosima::fastcdr::CdrVersion::XCDRv1);  // Object that serializes the data.
```
Comment (EduPonz) [inline rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:305]: Should we set the enconding flag here?

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp [hunk 2]
```diff
@@ -377,12 +378,12 @@ const TypeObject * GetMinimalObject(
...
```
Comment (EduPonz) [inline rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:381]: Should we set the enconding flag here?

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp (hunk 1 above)
Comment (MiguelCompany) [reply rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:305]: This is following what is done by Fast DDS Gen [here](https://github.com/eProsima/Fast-DDS-Gen/blob/master/src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectSource.stg#L226-L229), so I don't know.

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp (hunk 2 above)
Comment (MiguelCompany) [reply rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:381]: This is following what is done by Fast DDS Gen [here](https://github.com/eProsima/Fast-DDS-Gen/blob/master/src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectSource.stg#L169-L172), so I don't know.

File: rmw_fastrtps_dynamic_cpp/CMakeLists.txt [hunk 3]
```diff
@@ -38,7 +38,7 @@ find_package(rmw_dds_common REQUIRED)
 find_package(rmw_fastrtps_shared_cpp REQUIRED)
 
 find_package(fastrtps_cmake_module REQUIRED)
-find_package(fastcdr REQUIRED CONFIG)
+find_package(fastcdr 2 REQUIRED CONFIG)
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply rmw_fastrtps_dynamic_cpp/CMakeLists.txt:41]: 9687925

File: rmw_fastrtps_shared_cpp/package.xml [hunk 4]
```diff
@@ -21,7 +21,7 @@
   <buildtool_export_depend>ament_cmake</buildtool_export_depend>
 
   <build_depend>rosidl_dynamic_typesupport</build_depend>
-  <build_depend>fastcdr</build_depend>
+  <build_depend version_gte="2">fastcdr</build_depend>
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply rmw_fastrtps_shared_cpp/package.xml:24]: 9687925

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp (hunk 1 above)
Comment (EduPonz) [reply rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:305]: I'd like @richiware's input on this

Comment (richiware) [reply rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:305]: Fast DDS Gen doesn't set the encoding flag because Fast CDR uses PLAIN_CDR as default.

Comment (MiguelCompany) [reply rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:305]: Done

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp (hunk 2 above)
Comment (MiguelCompany) [reply rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:382]: This one also

**Review by EduPonz — APPROVED** [review]

**Pull request closed without merging**